inspirehep/modules/multieditor/api.py

```diff
@@ -0,0 +1,160 @@
+    if multieditor_session:
+        ids = multieditor_session['ids']
+        if all_selected:
+            ids = set(ids) - set(request.json['ids'])
+        else:
+            ids = set(ids) & set(request.json['ids'])
+    else:
+        return jsonify({'message': 'Please use the search before you apply actions'}), 400
+    actions = []
+    conditions = sanitize_user_conditions(user_actions.get('conditions', []))
+
+    for action in user_actions.get('actions', []):
+        actions.append(
+            ActionFactory.factory(action.get('actionName', ''),
+                                  **sanitize_user_action(action, conditions))
+        )
+    for i, chunk in enumerate(chunker(ids, 20)):
+        tasks.process_records.delay(records_ids=chunk, actions=actions, schema=multieditor_session['schema'])
+    return jsonify({'message': 'Records are being updated'})
+
+
+@blueprint.route("/preview", methods=['POST'])
+@multieditor_use_permission.require(http_exception=403)
+def preview():
+    """Preview the user actions in the first (page size) records."""
+    user_actions = request.json['userActions']
+    query_string = request.json['queryString']
+    page_size = int(request.json['pageSize'])
+    page_num = request.json['pageNum']
+    multieditor_session = session.get('multieditor_session', [])
+
+    if not multieditor_session:
+        return jsonify({'message': 'Please use the search before you apply actions'}), 400
+
+    records = queries.get_records_from_query(query_string, page_size, page_num, multieditor_session['schema_name'])['json_records']
+
+    old_records = copy.deepcopy(records)
+    actions = []
+    conditions = sanitize_user_conditions(user_actions.get('conditions', []))
+
+    for action in user_actions.get('actions', []):
+        actions.append(
+            ActionFactory.factory(action.get('actionName', ''),
+                                  **sanitize_user_action(action, conditions))
+        )
+    for record in records:
+        for action in actions:
+            action.apply(record, multieditor_session['schema'])
+
+    json_patches, errors = compare_records(old_records, records, multieditor_session['schema'])
+
+    return jsonify({'json_records': records, 'errors': errors, 'json_patches': json_patches})
+
+
+@blueprint.route("/search", methods=['GET'])
+@multieditor_use_permission.require(http_exception=403)
+def search():
+    """Search for records using the query and store the result's ids"""
+    query_string = request.args.get('queryString', '')
+    page_num = int(request.args.get('pageNum', 1))
+    page_size = int(request.args.get('pageSize', 1))
+    schema_name = request.args.get('index', '')
+    schema = load_schema(schema_name, resolved=True)
+    paginated_records = queries.get_records_from_query(query_string, page_size, page_num, schema_name)
+    if paginated_records['total_records'] > 10000:
+        return jsonify({'message': 'Please narrow the results using a query'}), 400
```

Comment:
And in this message if you show it in the UI, indicate that search results should be less than `10000`